docs: Remove refs of the NODE_ENV (#1400)
cube-js/cube.js: docs/Deployment/Production-Checklist.md

This is a checklist for configuring and securing Cube.js for a production
deployment.

> ## Disable Dev Mode

Reviewer: I think that, generally, we should refrain from using contractions because they decrease the accessibility of our docs. It's okay to have `DEV` in the code though.
```suggestion
## Disable Development Mode
```

> When running Cube.js in production make sure dev mode is disabled. Running dev
> mode in production environment can lead to security vulnerabilities.

Reviewer: I doubt that using "security vulnerabilities" here is appropriate because it really depends on the software using Cube.js and the nature of the data. IMO, it's quite misleading right now.

I think we can be more specific and add a relevant link here:

```suggestion
When running Cube.js in production environment, make sure the development mode is disabled. Running Cube.js in development
mode in a production environment allows accessing Cube.js API [without authorization](https://cube.dev/docs/security).
```

> The dev mode is disabled by default.

Reviewer: ```suggestion
The development mode is disabled by default.
```
